editor/plugins/curve_editor_plugin.cpp

```diff
@@ -704,70 +704,54 @@ Vector2 CurveEdit::get_tangent_view_pos(int p_index, TangentIndex p_tangent) con
 	return tangent_view_pos;
 }
 
-Vector2 CurveEdit::get_view_pos(Vector2 p_world_pos) const {
+Vector2 CurveEdit::get_view_pos(const Vector2 &p_world_pos) const {
 	return _world_to_view.xform(p_world_pos);
 }
 
-Vector2 CurveEdit::get_world_pos(Vector2 p_view_pos) const {
+Vector2 CurveEdit::get_world_pos(const Vector2 &p_view_pos) const {
 	return _world_to_view.affine_inverse().xform(p_view_pos);
 }
 
 // Uses non-baked points, but takes advantage of ordered iteration to be faster.
-template <typename T>
-static void plot_curve_accurate(const Curve &curve, float step, Vector2 scaling, T plot_func) {
-	if (curve.get_point_count() <= 1) {
-		// Not enough points to make a curve, so it's just a straight line.
-		// The added tiny vectors make the drawn line stay exactly within the bounds in practice.
-		float y = curve.sample(0);
-		plot_func(Vector2(0, y) * scaling + Vector2(0.5, 0), Vector2(1.f, y) * scaling - Vector2(1.5, 0), true);
+void CurveEdit::plot_curve_accurate(float p_step, const Color &p_line_color, const Color &p_edge_line_color) {
+	if (curve->get_point_count() <= 1) { // Draw single line through entire plot
+		float y = curve->sample(0);
+		draw_line(get_view_pos(Vector2(0.f, y)), get_view_pos(Vector2(1.f, y)), p_line_color, LINE_WIDTH, true);
```

Comment:
You shouldn't remove the little vector tweaks, they prevent this

![image](https://github.com/godotengine/godot/assets/85438892/17908bb7-a966-429f-9a04-ff7e4d9d00bd)

which is literally unplayable!